packages/engine-server/src/workspace/utils.ts

```diff
@@ -57,4 +68,97 @@ export class WorkspaceUtils {
     }
     return noteHash !== note.contentHash;
   }
+
+  /**
+   * Generate url for given note or return `undefined` if no url is specified
+   * @param opts
+   *
+   */
+  static getNoteUrl(opts: {
+    config: DendronConfig;
+    note: NoteProps;
+    vault: DVault;
+    urlRoot?: string;
```

Comment:
Seems like urlRoot is determined from within DendronConfig.  Should we remove this parameter and move the `const urlRoot = dendronConfig.site?.siteUrl || "";` logic into this function?

Reply:
We could, but in `CopyNoteUrl` it is also determined from `DendronWorkspace`
```ts
const urlRoot =
      config?.site?.siteUrl ||
      DendronWorkspace.configuration().get<string>(
        CONFIG.COPY_NOTE_URL_ROOT.key
      );
```